Akka.Persistence.Pulsar/Akka.Persistence.Pulsar.csproj

```diff
@@ -7,7 +7,15 @@
 
     <ItemGroup>
       <PackageReference Include="Akka.Persistence.Query" Version="1.4.0-beta3" />
+      <PackageReference Include="CassandraCSharpDriver" Version="3.12.0" />
```

Comment:
What do you need Cassandra for?

Reply:
To keep track of MessageId. It could be any storage,  just created a default implementation  of IMetadataStore in Cassandra! MessageId is the responsibility of pulsar server and we can only store and reuse.

Reply:
I will move IMetadataStore implementation to the client so avoid depending on/referencing Cassandra

Reply:
Doesn't this defeat the purpose of using Pulsar as an event journal? If every persisted event must be reflected in persisting corresponding record in another database (which will be probably a speed-constraining factor), then why use Pulsar at all?

Reply:
I will think on that and see if that can be done entirely in Pulsar.
Pulsar SQL would have been a good option but it requires Presto!

Consumer/Reader connect to different ledger once in while - the result being unreliable replay of messages from a supplied MessageId!

Since we are not in charge of generating MessageId, what can be done? Not like generating primary key from application code!

The problem I was trying to solve is getting the current MessageId from the supplied SequenceNr and PersistenceId.

Reply:
Maybe there's a possibility to add some metadata to Pulsar events? I'll dig more into pulsar soon, but the [documentation](https://pulsar.apache.org/docs/v2.0.1-incubating/project/BinaryProtocol/) sounds like it should be doable, as it states that message metadata contains:

> **sequence_id** The sequence ID of the message, assigned by producer

Reply:
I read somewhere that the sequence_id assigned by the producer is for de-duplication purposes. It does not help when we need StartMessageId.

Pulsar SQL seems the best option right now but we are lacking one thing to make it work which is Schema registration!

I will attempt to see If I am able to port it to C# for use in dotPulsar. If you know anyone, though, who can do it easily and has the time, I wont mind - am frightened ;)

Reply:
I have created a private(while it is being cleansed) repo and I have sent you an invite. 
Please take a look!